fix(astro:content): add deprecated to `getEntryBySlug` & `getDataEntryById` (#11600)
withastro/astro: .changeset/dull-taxis-warn.md

---
'astro': patch
---

> Adds deprecation warning on `getEntryBySlug` and `getDataEntryById` functions exported by `astro:content`.

Author: After thinking about it, maybe this would be better:
```suggestion
Deprecates `getEntryBySlug` and `getDataEntryById` functions exported by `astro:content` in favor of `getEntry`.
```
This would let users know directly that they should look for `getEntry`.

Reviewer: Yes, I like this suggestion!
